base/server/python/pki/server/cli/acme.py

```diff
@@ -687,11 +690,24 @@ def execute(self, argv):
             sys.exit(1)
 
         instance_name = 'pki-tomcat'
+        database_type = None
+        props = {}
 
         for o, a in opts:
             if o in ('-i', '--instance'):
                 instance_name = a
 
+            elif o == '--type':
+                database_type = a
+                if database_type not in DATABASE_TYPES.values():
+                    raise Exception('Invalid database type: %s' % database_type)
+
+            elif o == '-D':
```

Comment:
May be I'm lost. You are looking for `-D`, but the input provided by the user will be of format `-Dusername=password`. Will this if condition be able to pick it up?

Why are we introducing a new type of option: `-D<key>=<value>`?

Reply:
The `-D` option is defined earlier:
```
opts, _ = getopt.gnu_getopt(argv, 'i:vD:', [
```
so getopt can actually recognize the `-D` option and return the `<key>=<value>` as the value of that option. The subsequent code will split the `<key>` and `<value>`.

This option is added to provide a way to configure the database/issuer silently (i.e. without an interactive mode). See the updated docs. For example, if you're configuring a PostgreSQL database and you use the default database name (i.e. acme) and default username (i.e. acme) but only the password is different, you only need to specify the password:
```
pki-server acme-database-mod --type postgresql \
    -Dpassword=otherpassword
```

Reply:
Thanks for the details. I read the docs, but, I was curious (and a bit nit picky) on why we changed from our mainstream. 
````
$ pki-server acme-database-mod --type postgresql \
    -D user=admin \
    -D password=somepassword
````

Reply:
Both formats are accepted by getop. Not sure what is considered mainstream here, but in other tools the `-D` is usually used without a space:
https://www.rapidtables.com/code/linux/gcc/gcc-d.html
https://cmake.org/cmake/help/v3.0/manual/cmake.1.html
https://docs.oracle.com/en/java/javase/13/docs/specs/man/java.html